packages/ng-table-builder/integration/tests/samples/lifecycle/lifecycle.spec.ts

```diff
@@ -114,19 +119,37 @@ describe('[TEST]: Lifecycle table', () => {
         table.columnList = new QueryList<ElementRef<HTMLDivElement>>();
     });
 
-    it('should be basic api', () => {
-        table.source = JSON.parse(JSON.stringify(data));
+    it('should be basic api', async () => {
+        table.setSource(JSON.parse(JSON.stringify(data)));
 
+        expect(new TableSelectedItemsPipe(table).transform({ 1: true, 2: true })).toEqual([
+            { position: 1, name: 'Hydrogen', symbol: 'H', weight: 1.0079 },
+            { position: 2, name: 'Helium', weight: 4.0026, symbol: 'He' }
+        ]);
+
+        // enable and check filter
+        table.enableFiltering = true;
+        table.filterable.filterValue = 'Hydrogen';
+        table.filterable.filterType = 'START_WITH';
+        await table.sortAndFilter();
+
+        expect(table.source).toEqual([{ name: 'Hydrogen', position: 1, symbol: 'H', weight: 1.0079 }]);
+
+        // expect selection pipe works even with filtered values
         expect(new TableSelectedItemsPipe(table).transform({})).toEqual([]);
-        expect(new TableSelectedItemsPipe(table).transform({ 1: true })).toEqual([
-            {
-                name: 'Hydrogen',
-                position: 1,
-                symbol: 'H',
-                weight: 1.0079
-            }
+        expect(new TableSelectedItemsPipe(table).transform({ 1: true, 2: true })).toEqual([
+            { position: 1, name: 'Hydrogen', symbol: 'H', weight: 1.0079 },
+            { position: 2, name: 'Helium', weight: 4.0026, symbol: 'He' }
         ]);
 
+        expect(table.lastItem).toEqual({ position: 1, name: 'Hydrogen', symbol: 'H', weight: 1.0079 });
+
+        // disable filter
+        table.enableFiltering = true;
+        table.filterable.filterValue = null;
+        table.filterable.filterType = null;
+        await table.sortAndFilter();
```

Comment:
what do you think about create `await table.resetToDefaultFilter()`

Reply:
hello druzhische!
i don't think so. i guess it will be a function used only for tests. also i do not know what will be right to include into it, the model of filter is more complexive than i represented here. `filterValue` and `filterType` nicely updates by filter controls.